static/src/javascripts/projects/commercial/modules/sticky-top-banner.js

```diff
@@ -0,0 +1,210 @@
+                        stickyBanner.style.position = '';
+                        stickyBanner.style.top = '';
+                    }
+                }
+            })
+            .then(setupAnimation);
+    }
+
+    return Promise.resolve();
+};
+
+const update = (newHeight: number): Promise<any> =>
+    fastdom
+        .read(() => {
+            topSlotStyles = topSlotStyles || window.getComputedStyle(topSlot);
+            return (
+                newHeight +
+                parseInt(topSlotStyles.paddingTop, 10) +
+                parseInt(topSlotStyles.paddingBottom, 10)
+            );
+        })
+        .then(resizeStickyBanner);
+
+const onResize = (specs: any, _: any, iframe: ?Element): void => {
+    if (topSlot && topSlot.contains(iframe)) {
+        update(specs.height);
+        unregister('resize', onResize);
+    }
+};
+
+// Register a message listener for when the creative wants to resize
+// its container
+// We also listen for scroll events if we need to, to snap the slot in
+// place when it reaches the end of the header.
+const setupListeners = (): void => {
+    register('resize', onResize);
+    if (!config.page.hasSuperStickyBanner) {
+        addEventListener(window, 'scroll', onScroll, {
+            passive: true,
+        });
+    }
+};
+
+const getAdvertSizeByIndex = (advert: ?Advert, index: number): ?number => {
+    if (advert && advert.size && typeof advert.size !== 'string') {
+        return advert.size[index];
+    }
+};
+
+const onFirstRender = (): Promise<any> =>
+    trackAdRender(topSlotId).then(isRendered => {
+        if (isRendered) {
+            const advert = getAdvertById(topSlotId);
+            const adSize0 = getAdvertSizeByIndex(advert, 0);
+            const adSize1 = getAdvertSizeByIndex(advert, 1);
+
+            if (
+                // skip for Fabric creatives
+                adSize0 !== 88 &&
+                // skip for native ads
+                adSize1 &&
+                adSize1 > 0
+            ) {
+                return fastdom
+                    .read(() => {
+                        const styles = window.getComputedStyle(topSlot);
+
+                        return (
+                            parseInt(styles.paddingTop, 10) +
+                                parseInt(styles.paddingBottom, 10) +
+                                adSize1 || 0
+                        );
+                    })
+                    .then(resizeStickyBanner);
+            }
+            return fastdom
+                .read(() => {
+                    if (topSlot) {
+                        return topSlot.offsetHeight;
+                    }
+
+                    return 0;
+                })
+                .then(resizeStickyBanner);
+        }
+    });
+
+const initState = (): Promise<any> =>
+    fastdom
+        .read(() => {
+            if (header) {
+                headerHeight =
+                    parseInt(window.getComputedStyle(header).height, 10) || 0;
```

Comment:
I made this change, slightly grudgingly, because JSDOM doesn't support `offsetHeight`. It does feel wrong to change the implementation to make the tests pass. I'll try to find a better way of mocking `offsetHeight` without affecting the implementation.

As an aside, I believe the tests were originally failing intermittently because `offsetHeight` was sometimes still set to `0` at this point, possibly due to a race with async DOM mutations.